include/ensmallen_bits/cne/cne_impl.hpp

```diff
@@ -119,7 +121,11 @@ double CNE::Optimize(DecomposableFunctionType& function, arma::mat& iterate)
   // Set the best candidate into the network parameters.
   iterate = population.slice(index(0));
 
-  return function.Evaluate(iterate);
+  if (exactObjective)
+  {
+    lastBestFitness = function.Evaluate(iterate);
+  }
+  return lastBestFitness;
```

Comment:
Since CNE uses only an arbitrary function type, there's no need for `exactObjective` here.  It only applies to optimizers that optimize decomposable functions.  I think the documentation is a little wrong for CNE though (I updated it in another PR that I'm about to open).